src/cairoWriter/writers/cairoContractWriter.ts

```diff
@@ -71,9 +73,9 @@ export class CairoContractWriter extends CairoASTNodeWriter {
       .map((l) => (l.length > 0 ? INDENT + l : l))
       .join('\n');
 
-    const parentSourceUnit = node.parent;
-    assert(parentSourceUnit instanceof SourceUnit);
-    const otherStorageVars = parentSourceUnit.vFunctions.filter(
+    const sourceUnit = node.parent;
+    assert(sourceUnit instanceof SourceUnit, 'Contract node parent should be a Source Unit node.');
```

Comment:
Please use util function `getContainingSourceUnit`

```suggestion
const sourceUnit = getContainingSourceUnit(node);
```

Reply:
It will end up doing the same and this approach is a bit more strict.

Reply:
More strict in what sense?

Reply:
You make sure that the contract definition is a direct child of source unit

Reply:
Have it your way then, I think that assertions at this stage  (the last stage of the transpiler) are just a formality to satisfy TS's type checking. I don't think a "more" strict check at this stage will do any good